megatron/core/inference/engines/dynamic_engine.py

```diff
@@ -159,6 +161,11 @@ def __init__(
                 controller.inference_wrapped_model.model.config.enable_cuda_graph
             )
 
+        if process_group_collection is not None:
+            self.process_group_collection = process_group_collection
+        else:
+            self.process_group_collection = parallel_state
```

Comment:
Recommendation is use `ProcessGroupCollection.use_mpu_process_groups` to pick up global groups from parallel_state and create a pg_collection if not provided. And do it up in the call stack possible so lower stack doesn't need to change and be aware of whether it is customized PG or the conventional globally managed one.